middleware/src/main/java/br/ufrn/dimap/middleware/MiddlewareConfig.java

```diff
@@ -0,0 +1,163 @@
+		private static final Collection<InvocationInterceptorSerialized> serverRequestInteceptors = new ConcurrentLinkedQueue<>();
+		
+		/**
+		 * Registers an interceptor to be executed by the client on an
+		 * invocation. Happens before the object serialization.
+		 * 
+		 * @param interceptor interceptor to be registered
+		 */
+		public static void registerClientInvocationInterceptor(InvocationInterceptorUnserialized interceptor) {
+			clientInvocationInteceptors.add(interceptor);
+		}
+		
+		/**
+		 * Registers an interceptor to be executed by the server on an
+		 * invocation. Happens after the object deserialization.
+		 * 
+		 * @param interceptor interceptor to be registered
+		 */
+		public static void registerServerInvocationInterceptor(InvocationInterceptorUnserialized interceptor) {
+			serverInvocationInteceptors.add(interceptor);
+		}
+		
+		/**
+		 * Registers an interceptor to be executed by the client on an
+		 * invocation. Happens after the object serialization.
+		 * 
+		 * @param interceptor interceptor to be registered
+		 */
+		public static void registerClientRequestInterceptor(InvocationInterceptorSerialized interceptor) {
+			clientRequestInteceptors.add(interceptor);
+		}
+	
+		/**
+		 * Registers an interceptor to be executed by the client on an
+		 * invocation. Happens after the object serialization.
+		 * 
+		 * @param interceptor interceptor to be registered
+		 */
+		public static void registerServerRequestInterceptor(InvocationInterceptorSerialized interceptor) {
+			serverRequestInteceptors.add(interceptor);
+		}
+
+		/**
+		 * Returns a collection with the client invocation interceptors in the order
+		 * they must be applied
+		 * @return the interceptor collection
+		 */
+		public static Collection<InvocationInterceptorUnserialized> getClientinvocationinteceptors() {
+			return clientInvocationInteceptors;
+		}
+
+		/**
+		 * Returns a collection with the server invocation interceptors in the order
+		 * they must be applied
+		 * @return the interceptor collection
+		 */
+		public static Collection<InvocationInterceptorUnserialized> getServerinvocationinteceptors() {
+			return serverInvocationInteceptors;
+		}
+
+		/**
+		 * Returns a collection with the client request interceptors in the order
+		 * they must be applied
+		 * @return the interceptor collection
+		 */
+		public static Collection<InvocationInterceptorSerialized> getClientrequestinteceptors() {
+			return clientRequestInteceptors;
+		}
+
+		/**
+		 * Returns a collection with the server request interceptors in the order
+		 * they must be applied
+		 * @return the interceptor collection
+		 */
+		public static Collection<InvocationInterceptorSerialized> getServerrequestinteceptors() {
+			return serverRequestInteceptors;
+		}
+	}
+	
+	/**
+	 * This nested class holds the information regarding the protocolPlugins of the framework
+	 * 
+	 * @author Gustavo Carvalho
+	 */
+	public static class ProtocolPlugins {
```

Comment:
Consider changing this to a Singleton, instead of using static for all methods and attributes.